governance/api-reviews.md

```diff
@@ -0,0 +1,203 @@
+# tf-api-owners review practices
+
+## Overview
+
+This is an attempt to gather commonly discussed topics when doing tf-api-owners
+reviews. It’ll hopefully be a useful resource to both API owners and people
+proposing API changes.  Process TF API Owners meet twice weekly to discuss
+changes. We try to get to PRs on the next meeting, but we don’t always make it
+all the way through. If your change is particularly urgent, please ping the PR
+to notify us of any urgency.
+
+## Process
+
+We only look at changes which have already been approved by other reviewers. If
+there are major outstanding comments, we will wait with API review until those
+are resolved. If there are questions for API owners, explicitly raise this in
+the comments to get an answer.
+
+
+## High level points
+
+### Backward and forward compatibility
+We avoid backwards-incompatible API changes. We also avoid
+backwards-incompatible behavior changes, such as restricting the set of valid
+inputs to a function or extending the set of valid outputs of a function. Adding
+support for previously not supported behavior is okay, as are changes to
+explicitly experimental APIs (within reason). When needing to provide a new or
+different behavior, we strongly prefer a new version of the API over breaking
+backwards compatibility. Note that we are free to deprecate APIs; we just cannot
+break code which relies on their documented behavior. We need to worry about
+backward compatibility both of our python APIs and of the serialized GraphDefs,
+and in general breaking serialized GraphDefs is worse than breaking the python
+APIs.
+
+Forward compatibility is more subtle: we should avoid changing the graph
+produced by currently correct python code without a three weeks notice . This
+comes up most frequently when adding new ops, but also applies to non-obvious
+things such as the graph emitted by gradients or pfor.
+
+Including the name “experimental” in an API endpoint allows you to break
+backwards compatibility in the future, as noted above. However, we still prefer
+to mark the experimental API as deprecated for one release before removing it in
+the subsequent release. Please do not use the experimental namespace as an
+escape hatch for bad code or functionality you
+don’t-really-want-but-need-for-now; experimental APIs should be APIs that we
+intend to make standard in the near future, but have some lingering questions to
+resolve first.
+
+
+### Docstrings 
+
+TF APIs should have comprehensive documentation in the form of docstrings. If at
+all possible these docstrings should have runnable examples, and these examples
+should form a doctest so they stay correct. The examples should demonstrate an
```

Comment:
Can we link to this doc which describes it/gives more examples:

https://www.tensorflow.org/community/contribute/docs_ref

Reply:
Done